tests/test_samples.py

```diff
@@ -85,6 +85,24 @@ def test_bad_yaml():
     assert len(res) == 5
     assert not os.path.exists(file_path)
 
+
+def test_bad_yaml_2():
```

Comment:
@bhavikapanara we also need tests that actually trigger all the new exceptions that appear in user's error logs and make sure they are properly handled. This test only verifies for one exception. Does not test for the duplicated id0001 error. 

We also need a test to ensure that  the API works well when the event data is cleanly formatted after logging an event with numpy types.

Please take a closer look at the codecov report to see what other new code needs to be covered by tests.